File: backend/app/src/main/java/com/onair/hearit/app/recommendhearit/dto/RecommendHearitResponse.java
package com.onair.hearit.app.recommendhearit.dto;

import com.onair.hearit.core.domain.Hearit;
import java.time.LocalDateTime;

public record RecommendHearitResponse(
        Long id,
        String title,
        Integer playTime,
        LocalDateTime createdAt,
        String categoryName,
        String categoryColor
) {
    public static RecommendHearitResponse from(Hearit hearit) {
        return new RecommendHearitResponse(
                hearit.getId(),
                hearit.getTitle(),
                hearit.getPlayTime(),
                hearit.getCreatedAt(),
                hearit.getCategory().getName(),
                hearit.getCategory().getColorCode());
    }
}

File: android/app/src/main/java/com/onair/hearit/data/dto/RecommendHearitResponse.kt
package com.onair.hearit.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RecommendHearitResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("title")
    val title: String,
    @SerialName("playTime")
    val playTime: Int,
    @SerialName("createdAt")
    val createdAt: String,
    @SerialName("categoryName")
    val categoryName: String,
    @SerialName("categoryColor")
    val categoryColor: String,
)
